shared/redux/rtk-apis/classworks/classworks.api.ts

```diff
@@ -0,0 +1,24 @@
+import { projectApi } from "../api.config";
+import { CreateResourceDto, IClassworkResource } from "./classworks.interface";
+
+const classworksApi = projectApi.injectEndpoints({
+  endpoints: (builder) => ({
+    uploadResource: builder.mutation<IClassworkResource, CreateResourceDto>({
+      query: ({ file, classroomId, title, description }) => {
+        const formData = new FormData();
+        formData.append("file", file);
+        formData.append("title", title);
+        formData.append("description", description);
+
+        return {
+          url: `/classworks/${classroomId}/resources`,
```

Comment:
Let's follow REST principles

```suggestion
          url: `/classrooms/${classroomId}/resources`,
```